### DISCUSSION TITLE
I get this error Cannot read properties of undefined (reading 'height') at setNodeHeight

### DISCUSSION THREAD

**Discussion by joseDaKing** [diegomura/react-pdf#1599] [Q&A]
I get this error in my code and I don't why

error:

TypeError: Cannot read properties of undefined (reading 'height')
    at setNodeHeight

**Comment by joseDaKing** [comment]
The problem was just I accidentally passed in a string to a View component like this
```tsx
const Component = ({ name }) => {
    return (
        <>
            {name &&
            <Text>
                {name}
            </Text>}
        </>
    );
}
```

**Reply by jeetiss** [reply]
Hi @joseDaKing 

Which version are you use? is it general or dynamic component (with render prop)?

**Reply by joseDaKing** [reply]
I am using version ^2.0.21 and it is general

**Reply by jeetiss** [reply]
I dug into this problem, and here is what I found.

- react-pdf cannot work with strings without the Text component.
- js return the empty string as a falsy value in jsx condition `{name && <Text>{name}</Text>}` when name is empty `''`
- react-pdf tries to create this document and fails because of first point

to fix your problem just convert the first part of the condition to boolean with `!!` like `{!!name && <Text>{name}</Text>}`, or `{name !== '' && <Text>{name}</Text>}`

I added a detailed error message in #1604, could you please check that it is clear and understandable?
